enrichments/trace/internal/elastic/span.go

```diff
@@ -456,6 +462,29 @@ func (s *spanEnrichmentContext) setDestinationService(span ptrace.Span) {
 	}
 }
 
+func (s *spanEnrichmentContext) setInferredSpans(span ptrace.Span) {
+	spanLinks := span.Links()
+	childIDs := pcommon.NewSlice()
+	for i := 0; i < spanLinks.Len(); i++ {
+		spanLink := spanLinks.At(i)
+		spanID := spanLink.SpanID()
+		spanLink.Attributes().RemoveIf(func(k string, v pcommon.Value) bool {
+			switch k {
+			case "is_child", "elastic.is_child":
+				if v.Bool() && !spanID.IsEmpty() {
```

Comment:
[**For reviewers**] In APM-Server we add an empty string in the slice if span ID is empty ([ref](https://github.com/elastic/apm-data/blob/46f8e38486b772090fbf1fdccc8eeea13c513075/input/otlp/traces.go#L1299)), however, I didn't think it was important so didn't handle it here. Let me know if it needs to be added.

Reply:
> I don't think span links are allowed to have an empty span ID in OTLP?

I am not sure about this, I kept the span ID check for now, let me know if you think we should remove this.

Reply:
> In APM data we add an empty string in the slice if span ID is empty ([ref](https://github.com/elastic/apm-data/blob/46f8e38486b772090fbf1fdccc8eeea13c513075/input/otlp/traces.go#L1299)), however, I didn't think it was important so didn't handle it here

Just wanted to agree with you here. I'd keep the span ID check as is just as a safety measure.